src/Jackett.Common/Indexers/BB.cs

```diff
@@ -108,27 +108,38 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
                 searchStrings.Add((query.SanitizedSearchTerm + " " + string.Format("\"Season {0}\"", query.Season)).Trim());
 
             var categories = MapTorznabCapsToTrackers(query);
-            var request_urls = new List<string>();
+            var requestUrls = new List<string>();
 
             foreach (var searchString in searchStrings)
             {
                 var queryCollection = new NameValueCollection();
-                queryCollection.Add("action", "basic");
+                var argAction = "basic";
+                var argnameQuery = "searchstr";
+
+                // If only Freeleech enabled, switch to advanced search
+                if (configData.Freeleech.Value)
+                {
+                    argAction = "advanced";
+                    argnameQuery = "torrentname";
+                    queryCollection.Add("freeleech", "1");
+                }
+
+                queryCollection.Add("action", argAction);
```

Comment:
seems like this will be a problem. the "advanced" search `torrentname` arg only searches the album name for music